next-data/providers/blogData.ts

```diff
@@ -0,0 +1,46 @@
+import { cache } from 'react';
+
+import generateBlogData from '@/next-data/generators/blogData.mjs';
+import type { BlogDataRSC } from '@/types';
+
+const blogData = generateBlogData();
+
+const provideBlogData = cache(
+  async (category?: string): Promise<BlogDataRSC> => {
+    return blogData.then(({ posts, categories, pagination }) => {
+      const meta = { categories, pagination };
+
```

Comment:
It may be better to keep the default pagination value in a variable as `{ next: null, prev: null }` and use it from there

Reply:
Huhn what? the original pagination is just an array of numbers (all the available years)

Reply:
I explained it wrong, if we are not using pagination anywhere here, we are sending `{ next: null, prev: null }` for now. Actually, I thought of assigning this value to a variable and using it in one place. We will have to look at these methods during the adoption of the new layout anyway, I think it is more useful to look at them there

Reply:
We use it on `generateStaticParams` for the endpoint 👁️

Reply:
And yes, pagination will definitely change on the future. I hate this `/year-XXX` approach.

It is mostly done for Static Exports, because then it generates all the pagination pages at once. But still we could do it better.

Right now what happens is that we have a /en/blog/pagination markdwon file which is the actual page for pagination.

I feel that for the **future** we can simply disable pagination for static exports.

**How would it work?**

- on SSR/ISR environments: You have one page for each category, and the initial content for the page is loaded through `fetch`
  - The server component accepts query params, which we use for pagination `?page=X`
  - Then the ISR component loads for that category + page (`await getBlogData(category, page)`
  - Renders the Server Component, no need of being a Client One
- on SSG environments: The `await getBlogData` will return all the blog posts for each category, and the main blog page will return all blog posts.

Reply:
This also means that https://github.com/nodejs/nodejs.org/pull/6137/files#diff-1f78ec1e8badcdf54c89d1ec495a0040cf1abdb7d441e3285a5bf65f5b286b04R59-R66 and the other `@deprecated` things will be removed.

Yes, this means the old pagination pages will be broken, we can create redirects on redirects.json for all those pages for a while, until Google understands that the page moved permanently and removes it from SEO

i.e. `/blog/year-2023` redirects to `/blog?page=1`

Reply:
Thanks for the explanation. I have the same feelings about the `year-` approach, It will be great to make these arrangements with new layouts 🚀